### PR TITLE
[WIP] [Draft] Logs for .NET Framework

### LINKED ISSUE #2310: Enable Logging for .NET Framework applications
# Feature Request

**Are you requesting automatic instrumentation for a framework or library? Please describe.**

- Framework or library name : [`ILogger`]
- Library type: [Logging]
- Library version: [Framework 4.6.2+]

**Is your feature request related to a problem? Please describe.**
It's vital we have Logging support in our .NET Framework applications, our application can implement an ILogger solution, but we need them to be picked up by the instrumentation.

**Describe the solution you'd like**
Automatic support for ILogger in .NET Framework

**Describe alternatives you've considered**
Considering writing to a file and having the Collector tail the file to be added.

**Additional context**
Slack conversation: https://cloud-native.slack.com/archives/C01NR1YLSE7/p1678375665847919

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
## Why

Fixes #2310 

## What

* Enabled logs for `net461`
* Modified test application for logs, it work when executed as standalone. 

image

* Integration test fails with both console and web implementation. Work in progress to debug and fix an issue. 

## Checklist

- ~~[ ] `CHANGELOG.md` is updated.~~
- ~~[ ] Documentation is updated.~~
- ~~[ ] New features are covered by tests.~~

### CODE REVIEW

File: src/OpenTelemetry.AutoInstrumentation.Native/netfx_assembly_redirection.h [hunk 1]
```diff
@@ -35,6 +35,7 @@ void CorProfiler::InitNetFxAssemblyRedirectsMap()
         { L"OpenTelemetry.AutoInstrumentation", {0, 6, 0, 0} },
         { L"OpenTelemetry.Exporter.Console", {1, 0, 0, 0} },
         { L"OpenTelemetry.Exporter.OpenTelemetryProtocol", {1, 0, 0, 0} },
+        { L"OpenTelemetry.Exporter.OpenTelemetryProtocol.Logs", {1, 0, 0, 0} },
```
Comment (Kielek) [inline src/OpenTelemetry.AutoInstrumentation.Native/netfx_assembly_redirection.h:38]: Please remember to extend: 
- build structure tests file
- nuspec file

File: test/test-applications/integrations/TestApplication.Logs/TestApplication.Logs.csproj [hunk 2]
```diff
@@ -1,7 +1,21 @@
-<Project Sdk="Microsoft.NET.Sdk.Web">
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
 
   <PropertyGroup>
-    <TargetFrameworks>net7.0;net6.0</TargetFrameworks>
+    <TargetFrameworks>net7.0;net6.0;net462</TargetFrameworks>
+    <ImplicitUsings>disable</ImplicitUsings>
```
Comment (Kielek) [inline test/test-applications/integrations/TestApplication.Logs/TestApplication.Logs.csproj:5]: Can you remove this line?

**Review by Kielek — COMMENTED** [review]
I see that it is in draft mode, just scrolled through the code

File: test/IntegrationTests/LogTests.cs [hunk 3]
```diff
@@ -82,5 +82,105 @@ public void EnableLogsWithCLRAndHostingStartup()
...
+                var logsAsString = Convert.ToString(logRecord);
+                return logsAsString != null && logsAsString.Contains("TestApplication.Logs.Controllers.TestController");
+            });
+        }
+
+        SetEnvironmentVariable("OTEL_DOTNET_AUTO_TRACES_INSTRUMENTATION_ENABLED", "false");
+        SetEnvironmentVariable("OTEL_DOTNET_AUTO_LOGS_INCLUDE_FORMATTED_MESSAGE", includeFormattedMessage.ToString());
+        SetEnvironmentVariable("OTEL_DOTNET_AUTO_LOGS_INSTRUMENTATION_ENABLED", "true");
+        SetEnvironmentVariable("OTEL_DOTNET_AUTO_LOGS_ENABLED", "true");
+        SetEnvironmentVariable("OTEL_DOTNET_AUTO_NETFX_REDIRECT_ENABLED", "false");
```
Comment (pjanotti) [inline test/IntegrationTests/LogTests.cs:115]: Q: why is this necessary?

**Comment by rajkumar-rangaraj** [comment]
In the last SIG meeting, we decided to pause our work until a stable version of the `OpenTelemetry.Exporter.OpenTelemetryProtocol.Logs` package is released. This package may be retired, and its features could be incorporated into the `OpenTelemetry.Exporter.OpenTelemetryProtocol` package. Therefore, we will wait for the stable release from the OpenTelemetry .NET SDK before proceeding.
